app/js/sagas/conceptsSaga.js

```diff
@@ -1,25 +1,47 @@
 import {
   call,
   take,
+  takeEvery,
   fork,
   put,
+  cancel,
 } from 'redux-saga/effects';
 
-import { FETCH_LAB_CONCEPTS } from '../actions/actionTypes';
+import { FETCH_LAB_CONCEPTS, SET_CONCEPT_MEMBER } from '../actions/actionTypes';
 import labConceptsAction from '../actions/labConceptsAction';
 
 import conceptRest from '../rest/concept';
 
-export function* loadSetMemebers(member) {
-  const { uuid } = member;
+export function* loadMemebers(set) {
+  const { uuid } = set.member;
   const response = yield call(conceptRest.getSetMembers, uuid);
-  yield put(labConceptsAction.setMembers(response));
+  yield put(labConceptsAction.setMembers(response, set.count));
 }
 
-export function* setLabConcepts() {
-  const { payload: { data: { setMembers, answers } } } = yield take(`${FETCH_LAB_CONCEPTS}_SUCCESS`);
+export function* setConceptMemebers(action) {
+  const { payload: { data: { setMembers } } } = action;
+  const members = setMembers;
 
-  if (setMembers.length) {
-    yield setMembers.map(member => fork(loadSetMemebers, member));
+  if (members.length) {
+    let iterator = 0;
+    let forkedProcess;
+    yield put(labConceptsAction.setFetchStatus(true));
+    try {
+      while (members[iterator]) {
+        forkedProcess = yield fork(
+          loadMemebers, { member: members[iterator], count: iterator },
+        );
+        iterator += 1;
+      }
+    } finally {
+      const count = iterator - 1;
+      yield take(`${SET_CONCEPT_MEMBER}_${count}`);
```

Comment:
I may not be following this 100%, but so is intended to block until all the concept members have been fetched?  What happens if the fetches don't happen in order? Can we guarantee the one with the highest count happens last?

Reply:
@mogoodrich I think it's still fine if they don't happen in order, although it always does. I think we would need to write an error middleware soon, that way we could handle this better, especially if any of the call fails

Reply:
Sounds good!